src/features/leverage-tokens/components/leverage-token-mint-modal/index.tsx

```diff
@@ -179,6 +179,9 @@ export function LeverageTokenMintModal({
   // Derive expected tokens from preview data (no local state needed)
   const [transactionHash, setTransactionHash] = useState<`0x${string}` | undefined>(undefined)
   const [error, setError] = useState('')
+  // Store the floor minShares when user reaches confirm step
+  const [savedFloorMinShares, setSavedFloorMinShares] = useState<bigint | undefined>(undefined)
+  const [isRefreshingQuote, setIsRefreshingQuote] = useState(false)
```

Comment:
Do we need this extra state? Isn't this already provided by the useMintPlanPreview hook Is refreshing?